diff --git a/engine/platform/vendor/quickjs/quickjs.c b/engine/platform/vendor/quickjs/quickjs.c
index ca2079d..db2a014 100644
--- a/engine/platform/vendor/quickjs/quickjs.c
+++ b/engine/platform/vendor/quickjs/quickjs.c
@@ -2074,7 +2074,11 @@ static inline BOOL js_check_stack_overflow(JSRuntime *rt, size_t alloca_size)
 /* Note: OS and CPU dependent */
 static inline uintptr_t js_get_stack_pointer(void)
 {
+#ifdef _MSC_VER
+    return (uintptr_t)_AddressOfReturnAddress();
+#else
     return (uintptr_t)__builtin_frame_address(0);
+#endif
 }
 
 static inline BOOL js_check_stack_overflow(JSRuntime *rt, size_t alloca_size)
